sqlest/src/main/scala/sqlest/ast/Relation.scala

```diff
@@ -104,6 +104,14 @@ case class TableFunctionApplication[+T](
   def tableAlias = aliasedAs getOrElse tableName
 }
 
+/**
+ * A temporary table for the given select statement.
+ */
+case class TableFunctionFromSelect[A, R <: Relation](
+  select: Select[A, R],
+  aliasedAs: Option[String]
```

Comment:
Since we've made the user provide `.as()` we can probably just make this a `String`